File: app/src/main/java/com/codeartist/androidpractice/db/PropertyDao.java
package com.codeartist.androidpractice.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.codeartist.androidpractice.model.Property;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface PropertyDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(ArrayList<Property> property);

    @Query("DELETE FROM property_table")
    void deleteAll();

    @Query("SELECT * FROM property_table ORDER BY price ASC")
    LiveData<List<Property>> getProperty();
}

File: app/src/main/java/com/codeartist/androidpractice/db/PropertyDao.kt
package com.codeartist.androidpractice.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.codeartist.androidpractice.model.Property
import java.util.*

@Dao
interface PropertyDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(property: List<Property>)

    @Query("DELETE FROM property_table")
    suspend fun deleteAll()

    @get:Query("SELECT * FROM property_table")
    val property: LiveData<List<Property>>
}